### PR TITLE
[new release] caldav (0.2.1)

### PR DESCRIPTION
**Pull request by hannesm**
A CalDAV server

- Project page: <a href="https://github.com/roburio/caldav">https://github.com/roburio/caldav</a>
- Documentation: <a href="https://roburio.github.io/caldav/">https://roburio.github.io/caldav/</a>

##### CHANGES:

* Use Mirage-kv 6.0.0 API (roburio/caldav#32 @hannesm)
* Log more write errors (roburio/caldav#32 @hannesm)
* Refine Webdav_fs.batch to return a ('a, [> `Msg of string ]) result io, so
  it can fail (roburio/caldav#32 @hannesm)
* unikernel: use git-kv instead of irmin-mirage-git (roburio/caldav#32 @hannesm)

### CODE REVIEW

**Comment by hannesm** [comment]
CI failures are unrelated (same lwt issue as #22695), fine to merge :)

File: packages/caldav/caldav.0.2.1/opam [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+bug-reports: "https://github.com/roburio/caldav/issues"
+dev-repo: "git+https://github.com/roburio/caldav.git"
+tags: ["org:mirage" "org:robur"]
+doc: "https://roburio.github.io/caldav/"
+license: "ISC"
+
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
```
Comment (kit-ty-kate) [inline packages/caldav/caldav.0.2.1/opam:20]:
```suggestion
  ["dune" "runtest" "-p" name "-j" jobs] {with-test & os != "macos"} # Local network access is forbidden in the macos sandbox
```

**Comment by kit-ty-kate** [comment]
Thanks. Could you return the fix to the opam file upstream?

**Comment by hannesm** [comment]
Thanks for merging.

> Could you return the fix to the opam file upstream?

To be honest, I'm not sure. What is the state of "opam sandbox" and macOS? Should we really annotate random packages with "test not available on macos", just to remove that again in the future (once opam sandbox on macOS is working nicely)?

By the time, the opam sandbox on macOS is in better shape, will there be a cleanup and some added "&& opam < 2.y.z" added? Any guideline would be helpful.

Would a change also lead OCaml-CI to not execute the tests on macOS (if the answer is yes, I'd be in favour to keep the tests to be executed for the development repository).

**Comment by kit-ty-kate** [comment]
> To be honest, I'm not sure. What is the state of "opam sandbox" and macOS? Should we really annotate random packages with "test not available on macos", just to remove that again in the future (once opam sandbox on macOS is working nicely)?
> 
> By the time, the opam sandbox on macOS is in better shape, will there be a cleanup and some added "&& opam < 2.y.z" added? Any guideline would be helpful.

This is just the state of sandboxing on macOS in general. I don’t see the networking magically working in the next 10 years personally (Apple doesn’t care about bugs or missing features in their CLI tools and already told me they won’t do anything about it)

> Would a change also lead OCaml-CI to not execute the tests on macOS (if the answer is yes, I'd be in favour to keep the tests to be executed for the development repository).

No that’s only when the tests are ran through opam. OCaml-CI only runs dune.

**Comment by hannesm** [comment]
Thanks for your reply @kit-ty-kate. I'll push the change upstream.

**Pull request merged**